File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        valueToIndex = {}
        for i in range(len(nums)):
            if(nums[i] in valueToIndex):
                return [valueToIndex.get(nums[i]), i]
            else:
                valueToIndex.update({target - nums[i]: i})

        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        /*
        after watching solution algorithm
        */
        int left = 0;
        int right = numbers.length-1;
        int sum = numbers[left] + numbers[right];
        while(sum != target){
            if(sum > target){
                right--;
            }
            else if(sum < target){
                left++;
            }
            sum = numbers[left] + numbers[right];
        }
        int[] result = {left+1, right+1};
        return result;

    }
}
